lib/Mount/MountProvider.php

```diff
@@ -85,7 +85,19 @@ public function getFoldersForUser(IUser $user) {
 	public function getMountsForUser(IUser $user, IStorageFactory $loader) {
 		$folders = $this->getFoldersForUser($user);
 
-		return array_map(function ($folder) use ($user, $loader) {
+		/** @var Folder $folder */
+		$userHome = \OC::$server->getRootFolder()->getUserFolder($user->getUID());
```

Comment:
Doesn't this trigger cyclic things?

Reply:
Didn't seem so when testing, but indeed this seems like it could cause issues. I'll dig a bit deeper tomorrow.

Reply:
Looks fine, as the mountpoints are only setup once for the user. So when getRootFolder is called, initMountPoints will not execute again.

https://github.com/nextcloud/server/blob/1b85ef4bf2a18f7ee7bca7bcd3d7702362125d2e/lib/private/Files/Filesystem.php#L403

The issue I see would be that mount providers that are registered after the group folder provider don't have their mountpoints added to the user root at that point. For sharing mounts this is handled by manually forcing that mount provider to be the last mounted. https://github.com/nextcloud/server/blob/3ad6084891b88b3b7ef784bbf297b0a57b282d77/lib/private/Files/Config/MountProviderCollection.php#L93-L101

But I assume we should only rename regular users files anyway, so that shouldn't be an issue?

Reply:
Ok lets just try it then...